backend/de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/purchaseordercreation/localorder/PurchaseOrderAggregationKey.java

```diff
@@ -43,10 +44,12 @@
 	private final WarehouseId warehouseId;
 	private final BPartnerId vendorId;
 	private final ZonedDateTime datePromised;
+	private final Dimension dimension;
 
 	private static final Comparator<PurchaseOrderAggregationKey> COMPARATOR = Comparator.comparing(PurchaseOrderAggregationKey::getOrgId)
 			.thenComparing(PurchaseOrderAggregationKey::getWarehouseId)
 			.thenComparing(PurchaseOrderAggregationKey::getVendorId)
+			.thenComparing(PurchaseOrderAggregationKey::getDimension)
```

Comment:
dummy-question: why not compare by dimension last?

Reply:
Right. Moved it.

Reply:
Note: all the above were handled but I created another branch and PR: https://github.com/metasfresh/metasfresh/pull/10663